src/Frontend/src/composables/fileDownloadCreator.ts

```diff
@@ -1,15 +1,25 @@
-﻿export function useDownloadFile(text: string, fileType: string, fileName: string) {
-  const blob = new Blob([text], { type: fileType });
+﻿export function useDownloadFileFromString(text: string, fileType: string, fileName: string) {
+  const fileBlob = new Blob([text], { type: fileType });
+  const url = URL.createObjectURL(fileBlob);
+  downloadFile(url, fileType, fileName);
+}
+
+export async function useDownloadFileFromResponse(response: Response, fileType: string, fileName: string) {
+  const fileBlob = await response.blob();
+  const url = window.URL.createObjectURL(new Blob([fileBlob], { type: fileType }));
+  downloadFile(url, fileType, fileName);
+}
 
-  const a = document.createElement("a");
-  a.download = fileName;
-  a.href = URL.createObjectURL(blob);
-  a.dataset.downloadurl = [fileType, a.download, a.href].join(":");
-  a.style.display = "none";
-  document.body.appendChild(a);
-  a.click();
-  document.body.removeChild(a);
+function downloadFile(url: string, fileType: string, fileName: string) {
```

Comment:
i'm fairly certain there's a better way of doing this now - from memory this used to be a hack because mozilla didn't support the cleaner method

Reply:
happy to change it to a better way?

Reply:
disregard, i was thinking of download from in-memory, not from url